client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,64 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final RetentionConfig retentionConfig;
+
+    public enum RetentionConfig {
+        /**
+         * This {@link ReaderGroup} is not a subscriber and does not participate in Consumption Based Retention.
+         *
+         * See <a href="https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention">Consumption Based Retention</a>
+         *
+         */
+        NONE(false, false),
+
+        /**
+         * This {@link ReaderGroup} is a subscriber and can retain stream data using Consumption Based Retention. This will configure the
```

Comment:
Perhaps this part should be moved to the doc for the whole enum. Also rather than using "Consumption based retention" as a phrase which users probably don't know, explicitly link to the config value you are referring to.

Reply:
Noted. Not using "Consumption Based Retention" for explanation.